XamlControlsGallery/ControlPages/ListViewPage.xaml.cs

```diff
@@ -56,6 +83,242 @@ private void SelectionModeComboBox_SelectionChanged(object sender, SelectionChan
+        }
+
+        private async void Target_Drop(object sender, DragEventArgs e)
+        {
+            if (e.DataView.Contains(StandardDataFormats.Text))
+            {
+
+                var def = e.GetDeferral();
+                var s = await e.DataView.GetTextAsync();
+                var items = s.Split('\n');
+                foreach (var item in items)
+                {
+
+                    // Create Contact object from string, add to existing target ListView
+                    var info = item.Split(" ", 3);
+                    Contact temp = new Contact(info[0], info[1], info[2]);
+                    
+                    // Find the insertion index:
+                    Windows.Foundation.Point pos = e.GetPosition(DragDropListView2.ItemsPanelRoot);
+
+                    // Find height of a random item from the target list view
+                    ListViewItem sampleItem = (ListViewItem)DragDropListView.ContainerFromIndex(0);
+                    double itemHeight = sampleItem.ActualHeight + sampleItem.Margin.Top + sampleItem.Margin.Bottom;
+
+                    // Find index based on dividing number of items by height of each item
+                    int index = Math.Min(DragDropListView2.Items.Count - 1, (int)(pos.Y / itemHeight));
+
+                    // Find the item that we want to drop
+                    ListViewItem targetItem = (ListViewItem)DragDropListView2.ContainerFromIndex(index); ;
+
+                    // Figure out if to insert above or below
+                    Windows.Foundation.Point positionInItem = e.GetPosition(targetItem);
+                    if (positionInItem.Y > itemHeight / 2)
+                    {
+                        index++;
+                    }
+
+                    // Don't go out of bounds
+                    index = Math.Min(DragDropListView2.Items.Count, index);
+
+                    // Finally, add to target listview
+                    contacts2.Insert(index, temp);
+
+                    // Go through source list and remove the items that are being moved
+                    foreach (Contact contact in DragDropListView.Items)
+                    {
+                        if (contact.FirstName == temp.FirstName && contact.LastName == temp.LastName && contact.Company == temp.Company)
+                        {
+                            contacts1.Remove(contact);
+                            break;
+                        }
+                    }
+                }
+                
+                e.AcceptedOperation = DataPackageOperation.Move;
+                def.Complete();
+            }
+        }
+
+        private async void Source_Drop(object sender, DragEventArgs e)
+        {
+
+            var def = e.GetDeferral();
+            var s = await e.DataView.GetTextAsync();
+            var items = s.Split('\n');
+            foreach (var item in items)
+            {
+                var info = item.Split(" ", 3);
```

Comment:
I would prefer to see real types instead of `var`. I think it makes the code easier to read by taking some of the mystery out of what's happening :)